manifests/quadlet.pp

```diff
@@ -83,9 +83,18 @@
 ) {
   $podman_version = fact('podman.version')
 
-  if $podman_version and versioncmp($podman_version, '4.4', true) >= 0 {
+  # If podman is not installed yet, require installation since the fact will not be populated yet
+  if $podman_version =~ Undef {
     require podman::install
+    $podman_supports_quadlets = true
+  } elsif versioncmp($podman_version, '4.4', true) >= 0 {
+    require podman::install
+    $podman_supports_quadlets = true
+  } else {
+    $podman_supports_quadlets = false
+  }
 
+  if $podman_supports_quadlets {
```

Comment:
I must admit I find the way to detect this (via a fact) rather error prone (as we see with the fix here).
And even in the case of the fix, nothing says that we *actually* have installed a 4.4+ (and thus support quadlets).

Wouldn't it be easier to say "Debian 13+, Ubuntu 24.04+, Fedora, Arch and EL8+ get quadlets, everyone else an error message"? No race conditions, just hard facts. (well, yes, on EL8 - and 8 only - there are ways to accidentally install an older version, but I think we can ignore these)

Reply:
I implemented this version of it.